gensim/models/keyedvectors.py

```diff
@@ -1130,16 +1274,24 @@ def evaluate_word_analogies(self, analogies, restrict_vocab=300000, case_insensi
                 self.vocab = ok_vocab
                 ignore = {a, b, c}  # input words to be ignored
                 predicted = None
-                # find the most likely prediction using 3CosAdd (vector offset) method
-                # TODO: implement 3CosMul and set-based methods for solving analogies
-                sims = self.most_similar(positive=[b, c], negative=[a], topn=5, restrict_vocab=restrict_vocab)
+                # find the most likely prediction using 3CosAdd (vector offset) method or 3CosMul method
+
+                if method == '3CosAdd':
+                    sims = self.most_similar(positive=[b, c], negative=[a], topn=topk, restrict_vocab=restrict_vocab)
+                elif method == '3CosMul':
+                    sims = self.most_similar_cosmul(positive=[b, c], negative=[a], topn=topk,
+                    restrict_vocab=restrict_vocab)
+                else:
+                    raise ValueError("Unknown method for the Evaluation on the Analogy test set...")
```

Comment:
This exception could be more helpful:

```suggestion
                    raise ValueError("expected method %r to be one of %r" % method, _ACCEPTED_METHODS)
```